docs/adr/0003-handle-client-configuration-via-connection-plugin.md

```diff
@@ -0,0 +1,22 @@
+# Why are we using a connection plugin to handle client configuration ?
+
+Date: 2023-02-07
+
+## Status
+
+Proposed
+
+## Context
+
+The Pravic project aims to manage resources on multiple cloud provider using the ``resource`` module.
+In order to create/delete/update theses resources, user need to provide YAML definition of the resources along with credentials to create cloud client configuration, so that the module can know if theses resources will be created on Kubernetes cluster, AWS or any other provider.
+
+Theses credentials can be provided as parameters of the ``resource`` module, this means for each provider on which we are going to deploy resources in we need a dedicated tasks with the corresponding parameters, this has the drawback to increase the complexity of the playbook and this can of playbook is not really reusable as this need to be adapted for each cloud provider on which we need to deploy resources.
```

Comment:
> we need a dedicated tasks with the corresponding parameters

The parameter itself can be the same across providers, though, which is how it currently works. There is a `connection` dictionary. The downside here is that the structure of that dictionary varies across providers, so we can't define that in a way that would make use of Ansible's builtin documentation tools and parameter validation.